File: web/src/main/java/com/isystk/sample/web/base/controller/api/resource/FieldErrorResource.java
package com.isystk.sample.web.base.controller.api.resource;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class FieldErrorResource {

  // 項目名
  String fieldName;

  // エラー種別
  String errorType;

  // エラーメッセージ
  String errorMessage;

  public FieldErrorResource() {
  }

  public String getFieldName() {
    return this.fieldName;
  }

  public String getErrorType() {
    return this.errorType;
  }

  public String getErrorMessage() {
    return this.errorMessage;
  }

  public void setFieldName(String fieldName) {
    this.fieldName = fieldName;
  }

  public void setErrorType(String errorType) {
    this.errorType = errorType;
  }

  public void setErrorMessage(String errorMessage) {
    this.errorMessage = errorMessage;
  }

  public boolean equals(final Object o) {
    if (o == this) {
      return true;
    }
    if (!(o instanceof FieldErrorResource)) {
      return false;
    }
    final FieldErrorResource other = (FieldErrorResource) o;
    if (!other.canEqual((Object) this)) {
      return false;
    }
    final Object this$fieldName = this.getFieldName();
    final Object other$fieldName = other.getFieldName();
    if (this$fieldName == null ? other$fieldName != null
        : !this$fieldName.equals(other$fieldName)) {
      return false;
    }
    final Object this$errorType = this.getErrorType();
    final Object other$errorType = other.getErrorType();
    if (this$errorType == null ? other$errorType != null
        : !this$errorType.equals(other$errorType)) {
      return false;
    }
    final Object this$errorMessage = this.getErrorMessage();
    final Object other$errorMessage = other.getErrorMessage();
    if (this$errorMessage == null ? other$errorMessage != null
        : !this$errorMessage.equals(other$errorMessage)) {
      return false;
    }
    return true;
  }

  protected boolean canEqual(final Object other) {
    return other instanceof FieldErrorResource;
  }

  public int hashCode() {
    final int PRIME = 59;
    int result = 1;
    final Object $fieldName = this.getFieldName();
    result = result * PRIME + ($fieldName == null ? 43 : $fieldName.hashCode());
    final Object $errorType = this.getErrorType();
    result = result * PRIME + ($errorType == null ? 43 : $errorType.hashCode());
    final Object $errorMessage = this.getErrorMessage();
    result = result * PRIME + ($errorMessage == null ? 43 : $errorMessage.hashCode());
    return result;
  }

  public String toString() {
    return "FieldErrorResource(fieldName=" + this.getFieldName() + ", errorType="
        + this.getErrorType() + ", errorMessage=" + this.getErrorMessage() + ")";
  }
}

File: web/src/main/kotlin/com/isystk/sample/web/base/controller/api/resource/FieldErrorResource.kt
package com.isystk.sample.web.base.controller.api.resource

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
class FieldErrorResource {
    // 項目名
    var fieldName: String? = null

    // エラー種別
    var errorType: String? = null

    // エラーメッセージ
    var errorMessage: String? = null
    override fun equals(o: Any?): Boolean {
        if (o === this) {
            return true
        }
        if (o !is FieldErrorResource) {
            return false
        }
        val other = o
        if (!other.canEqual(this as Any)) {
            return false
        }
        val `this$fieldName`: Any? = fieldName
        val `other$fieldName`: Any? = other.fieldName
        if (if (`this$fieldName` == null) `other$fieldName` != null else `this$fieldName` != `other$fieldName`) {
            return false
        }
        val `this$errorType`: Any? = errorType
        val `other$errorType`: Any? = other.errorType
        if (if (`this$errorType` == null) `other$errorType` != null else `this$errorType` != `other$errorType`) {
            return false
        }
        val `this$errorMessage`: Any? = errorMessage
        val `other$errorMessage`: Any? = other.errorMessage
        return if (if (`this$errorMessage` == null) `other$errorMessage` != null else `this$errorMessage` != `other$errorMessage`) {
            false
        } else true
    }

    protected fun canEqual(other: Any?): Boolean {
        return other is FieldErrorResource
    }

    override fun hashCode(): Int {
        val PRIME = 59
        var result = 1
        val `$fieldName`: Any? = fieldName
        result = result * PRIME + (`$fieldName`?.hashCode() ?: 43)
        val `$errorType`: Any? = errorType
        result = result * PRIME + (`$errorType`?.hashCode() ?: 43)
        val `$errorMessage`: Any? = errorMessage
        result = result * PRIME + (`$errorMessage`?.hashCode() ?: 43)
        return result
    }

    override fun toString(): String {
        return ("FieldErrorResource(fieldName=" + fieldName + ", errorType="
                + errorType + ", errorMessage=" + errorMessage + ")")
    }
}